### ISSUE TITLE
Add a local param for blocksize 

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-core#993]
As proposed sync by @musalbas, we should add a local configuration that validators can use to specify how large the blocks they produce are. This value would still be capped by the block size param in the consensus parameters, but validators could only want to producer smaller blocks.

**Comment by rootulp** [comment]
Why would block producers want to produce smaller blocks?

**Comment by musalbas** [comment]
* If producing larger blocks results in consensus failures if the p2p network is not optimized enough yet.
* The network may not be used enough, such that transactions fees are low enough that there's a lot of spam transaction, that are not worth the extra storage/bandwidth/etc costs for validators, or such that they are creating an artificial over-supply of block data.
* As a sort of "friendly agreement" with the DA network if the network becomes very adopted before we could implementing pruning in celestia-node.

**Comment by rootulp** [comment]
Thanks for elaborating. In all of those scenarios I think we'll be incentivized to cut a new release of celestia-app with a lower consensus param for max block size. My concern is that asking validators to manually configure a local block size param may not get full adoption and thus may not completely resolve the scenarios you described.

**Comment by rootulp** [comment]
> couldn't that be done via governance since the blocksize is an on-chain governance param also?

Yes I think so. My concern with this issue is that it introduces another lever so if we implement https://github.com/celestiaorg/celestia-app/issues/1592 I think 3 factor's could impact a block proposer's block size:

1. local config max block size
2. consensus param max block size
3. hard-coded max block size

and I don't yet understand why all 3 are necessary.

> I'm also open to just starting with a lower consensus block size on mainnet, instead of relying local config / etiquette.

Makes sense and seems like a safer option.
